### DISCUSSION TITLE
msw compatible with legacy edge 18

### DISCUSSION THREAD

**Discussion by harshan007** [mswjs/msw#1649] [Q&A]
We have a web app thats used in legacy edge 18 browser. We are setting up some browser compatibility tests in [BrowserStack](https://www.browserstack.com/automate).
 
To enable testing of UX pages, we use msw to return mocks from a API service. We use the latest version of msw and it doesn't work in legacy edge 18 to return mocks - although it does work in latest chrome, firefox, edge versions. Browserstack doesn't give any specific error but I could see from logs that its not even able to launch the page we feed in. It does work in other latest browsers.

  ```
const { worker } = require("./mocks/browser");
  ...
  void worker.start({ serviceWorker: { url } });

```
Looking for a old version of msw that is compatible with legacy edge 18?

**Comment by sharutkarsh** [comment]
Hi 

Could you please make sure to run the same test on a your real local device using a legacy edge 18 before testing on BrowserStack to make sure that it does work locally?

Given that MSW is not working in legacy Edge 18 and you require an older version that is compatible, I would suggest reaching out to checking the project's GitHub repository for any discussions or issues related to compatibility with legacy Edge 18. It's possible that someone in the community has found a workaround or developed a specific version that works with that particular browser.

Alternatively, you might consider exploring other mocking libraries or frameworks that specifically support legacy Edge 18.

I hope this helps.
